docs/pages/ds-video-port_L3_Low-Level_TestSpecification.md

```diff
@@ -0,0 +1,43 @@
+# Device Settings Video Port L3 Low Level Test Specification and Procedure Documentation
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [References](#references)
+- [Level 3 Test cases High Level Overview](#level-3-test-cases-high-level-overview)
+
+## Overview
+
+This document describes the L3 Low Level Test Specification and Procedure Documentation for the Device Settings Video Port module.
+
+### Acronyms, Terms and Abbreviations
+
+- `HAL` \- Hardware Abstraction Layer, may include some common components
+- `UT`  \- Unit Test(s)
+- `OEM` \- Original Equipment Manufacture
+- `SoC` \- System on a Chip
+- `HDMI`\- High-Definition Multimedia Interface
+- `HDCP`\- High-bandwidth Digital Content Protection
+- `HDR` \- High Dynamic Range
+- `HLG` \- Hybrid Log-Gamma
+- `SDR` \- Standard Dynamic Range
+
+### References
+
+- `High Level Test Specification` - [dsVideoPort High Level TestSpec](ds-video-port_High-Level_TestSpec.md)
+- `Interface header` - [dsVideoPort HAL header](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsVideoPort.h)
+
+## Level 3 Test Cases High Level Overview
```

Comment:
here's an overview of the module... Based on this list, we need to review what's the top 10 items of major items to test.

# Device Settings HAL Video Port Public API Functions

## Table 1: Initialization and Basic Port Management

| Function                      | Sink Device | Source Device | Description                                                                                                  |
| ----------------------------- | ----------- | ------------ | ------------------------------------------------------------------------------------------------------------ |
| `dsVideoPortInit()`           | Supported   | Supported    | Initializes the video port subsystem.                                                                         |
| `dsVideoPortTerm()`           | Supported   | Supported    | Terminates the video port subsystem and releases resources.                                                |
| `dsGetVideoPort()`            | Supported   | Supported    | Retrieves a handle to a specific video port.                                                                |
| `dsIsVideoPortEnabled()`      | Supported   | Supported    | Checks if a video port is enabled.                                                                          |
| `dsIsVideoPortActive()`       | Returns true (display always connected) | Checks if a video port is active (connected and in use). |
| `dsEnableVideoPort()`         | Supported   | Supported    | Enables or disables a video port.                                                                            |
| `dsIsDisplayConnected()`      | Returns true (display always connected) | Checks if a display is connected to the video port.              |

## Table 2: Display Capabilities and Settings

| Function                      | Sink Device | Source Device | Description                                                                                          |
| ----------------------------- | ----------- | ------------ | ---------------------------------------------------------------------------------------------------- |
| `dsIsDisplaySurround()`      | Supported   | Not Supported | Checks if the connected display supports audio surround sound.                                   |
| `dsGetTVHDRCapabilities()`    | Supported   | Not Supported | Gets the HDR capabilities of the connected TV/display.                                             |
| `dsSupportedTvResolutions()`  | Supported   | Not Supported | Gets the supported resolutions of the connected TV/display.                                         |
| `dsIsOutputHDR()`             | Supported   | Supported    | Checks if the video output is HDR.                                                                   |
| `dsGetCurrentOutputSettings()`| Supported   | Supported    | Gets multiple output settings (color space, color depth, etc.) in one call.                         |
| `dsGetVideoEOTF()`            | Supported   | Supported    | Gets the current HDR format (Electro-Optical Transfer Function) on a video port.                   |
| `dsGetMatrixCoefficients()`   | Supported   | Supported    | Gets the current matrix coefficients value of the video port.                                      |
| `dsGetColorDepth()`           | Returns default color depth | Gets the color depth of the video port.                               |
| `dsGetColorSpace()`           | Returns default color space | Gets the color space setting of the video port.                      |
| `dsGetQuantizationRange()`    | Supported   | Supported    | Gets the quantization range of the video port.                                                      |

## Table 3: Resolution and HDR Control (Primarily Source Devices)

| Function                | Sink Device | Source Device | Description                                                                                                  |
| ----------------------- | ----------- | ------------ | ------------------------------------------------------------------------------------------------------------ |
| `dsSetResolution()`      | Not Supported | Sets the display resolution for a video port.                                                        |
| `dsGetResolution()`      | Supported   | Gets the current display resolution of a video port.                                                 |
| `dsResetOutputToSDR()`  | Not Supported | Resets the video output to SDR.                                                                        |
| `dsSetForceHDRMode()`    | Not Supported | Sets/resets the force HDR mode.                                                                         |
| `dsGetSurroundMode()`    | Not Supported | Gets the surround mode of the video port.                                                               |
| `dsSetBackgroundColor()` | Not Supported | Sets the background color of the video port.                                                           |

## Table 4: HDCP Management (Primarily Source Devices)

| Function                        | Sink Device                | Source Device | Description                                                                                                           |
| ------------------------------- | -------------------------- | ------------ | --------------------------------------------------------------------------------------------------------------------- |
| `dsEnableHDCP()`               | Not Supported              | Enables/disables HDCP for a video port.                                                                      |
| `dsIsHDCPEnabled()`            | Returns true (always protected) | Indicates if a video port is HDCP protected.                                                              |
| `dsGetHDCPStatus()`             | Returns `dsHDCP_STATUS_AUTHENTICATED` | Gets the current HDCP status of the video port.                                                         |
| `dsGetHDCPProtocol()`           | Supported                  | Gets the HDCP protocol version of the device.                                                                 |
| `dsGetHDCPReceiverProtocol()`  | Not Supported              | Gets the HDCP protocol version of the connected sink device.                                               |
| `dsGetHDCPCurrentProtocol()`    | Supported                  | Gets the current negotiated HDCP protocol version.                                                        |
| `dsRegisterHdcpStatusCallback()`| Triggers callback once on bootup with `dsHDCP_STATUS_AUTHENTICATED` | Registers a callback to be notified of HDCP status changes.                                    |
| `dsSetHdmiPreference()`         | Supported                  | Sets the preferred HDMI Protocol of the specified video port                                              |
| `dsGetHdmiPreference()`         | Supported                  | Gets the preferred HDMI Protocol version of the specified video port                                      |

## Table 5: 4K and ALLM Control (Source Devices Only)

| Function                      | Sink Device | Source Device | Description                                                                                                     |
| ----------------------------- | ----------- | ------------ | --------------------------------------------------------------------------------------------------------------- |
| `dsSetForceDisable4KSupport()` | Not Supported | Not Supported (currently) | Sets a flag to forcefully disable 4K support.                                                                |
| `dsGetForceDisable4KSupport()` | Not Supported | Not Supported (currently) | Gets the status of the 4K support disable flag.                                                               |
| `dsSetAllmEnabled()`          | Not Supported | Enables/disables ALLM (Auto Low Latency Mode) for HDMI output.                                            |
| `dsGetAllmEnabled()`          | Not Supported | Checks if ALLM is enabled for an HDMI output.                                                               |

## Table 6: EDID and Color Depth Management (Source Devices Only)

| Function                 | Sink Device | Source Device | Description                                                                                                  |
| ------------------------ | ----------- | ------------ | ------------------------------------------------------------------------------------------------------------ |
| `dsGetIgnoreEDIDStatus()`| Not Supported | Gets whether to ignore EDID data.                                                                        |
| `dsColorDepthCapabilities()` | Not Supported | Gets the color depth capabilities of the video port                                                       |
| `dsGetPreferredColorDepth()` | Not Supported | Gets the preferred color depth of the video port                                                          |
| `dsSetPreferredColorDepth()` | Not Supported | Sets the preferred color depth for the video port                                                         |

## Table 7: Callback Registration

| Function                        | Sink Device | Source Device | Description                                                                                                   |
| ------------------------------- | ----------- | ------------ | ------------------------------------------------------------------------------------------------------------ |
| `dsVideoFormatUpdateRegisterCB()`| Supported   | Supported    | Registers a callback to be notified of video format update events.                                       |

**Grouping Rationale**

The functions are grouped into tables based on their primary functionality to improve readability and understanding.

This table format provides a clear overview of the API's capabilities and how they differ between sink and source devices.

**Please note that some functions may have additional nuances or dependencies not fully captured in this table. Always refer to the detailed comments within the `dsVideoPort.h` file for complete information**

Reply:
@Ulrond , we shall go ahead with the above approach and will not follow https://github.com/rdkcentral/rdk-halif-test-device_settings/pull/102#discussion_r1745284292. is that ok?